### PR TITLE
Use Anaconda for Python build pack

### PR DESCRIPTION
**Pull request by johnewart**
More reliable and faster than building from source -- thoughts?

### CODE REVIEW

**Comment by johnewart** [comment]
Also a minor bug fix for `exec` so that if the dependency setup fails then `exec` fails

File: buildpacks/python.go [hunk 1]
```diff
@@ -33,119 +32,120 @@ func (bt PythonBuildTool) Version() string {
...
+			return err
 		}
-	}
 
-	if _, err := os.Stat(pythonVersionDir); err == nil {
-		fmt.Printf("Python %s installed in %s\n", bt.Version(), pythonVersionDir)
-	} else {
-		os.Setenv("PYENV_ROOT", pyenvDir)
-		PrependToPath(filepath.Join(pyenvDir, "bin"))
+		// TODO: Windows
```
Comment (beholders-eye) [inline buildpacks/python.go]: I think we could trigger `Linux for Windows Subsystem` to test this, we can open another issue for that.

File: buildpacks/python.go [hunk 2]
```diff
@@ -33,119 +32,120 @@ func (bt PythonBuildTool) Version() string {
...
+	if _, err := os.Stat(envDir); err == nil {
+		fmt.Printf("environment installed in %s\n", envDir)
+	} else {
+		currentPath := os.Getenv("PATH")
+		newPath := fmt.Sprintf("PATH=%s:%s", filepath.Join(condaDir, "bin"), currentPath)
+		setupDir := bt.spec.PackageDir
+		condaBin := filepath.Join(condaDir, "bin", "conda")
+
+		for _, cmd := range []string{
+			fmt.Sprintf("%s config --set always_yes yes --set changeps1 no", condaBin),
```
Comment (beholders-eye) [inline buildpacks/python.go]: I'm not sure if this commands should be used every time to setup Anaconda/Miniconda

Comment (johnewart) [reply buildpacks/python.go]: It can't hurt; it does ensure that the config is properly set

Comment (johnewart) [reply buildpacks/python.go]: Miniconda does provide a python binary, we then install the requested python in an environment and use that.

File: buildpacks/python.go (hunk 1 above)
Comment (johnewart) [reply buildpacks/python.go]: Not everyone uses WSL, we should make sure we have an install / setup that works on pure Windows :)

**Comment by nictuku** [comment]
> If miniconda does provide a `python` executable, it would be a good substitute for `pyenv`. But I think it would be good to test more then one Python version, like 3.7.4 or 3.6.9.

Meta-point: Was this comment meant to block the PR? It doesn't seem actionable. When reviewing PRs, let's try to unblock the author ASAP. 👍

**Pull request merged**